This cannot be considered complete yet but it appears to work for
sile at least. There is an issue in the checkPhase that has something to do with libtexpdf that seems unrelated but I'm still digging into it.

```
/nix/store/kmhw99yxbwgqai9v8b3rwb4nf31z1cml-lua-5.2.4/bin/lua: error loading module 'justenoughlibtexpdf' from file './core/justenoughlibtexpdf.so':
	dlopen(./core/justenoughlibtexpdf.so, 0x0006): Library not loaded: /nix/store/q97r3hiic0jg6ks95xylazf9m21hd1wh-sile-0.12.5/lib/libtexpdf.0.dylib
  Referenced from: /Users/zaphar/projects/nixpkgs/sile-0.12.5/core/justenoughlibtexpdf.so
  Reason: tried: '/nix/store/q97r3hiic0jg6ks95xylazf9m21hd1wh-sile-0.12.5/lib/libtexpdf.0.dylib' (no such file), '/usr/local/lib/libtexpdf.0.dylib' (no such file), '/usr/lib/libtexpdf.0.dylib' (no such file)
stack traceback:
	[C]: in ?
	[C]: in function 'require'
	./core/libtexpdf-output.lua:1: in main chunk
	[C]: in function 'require'
	./core/sile.lua:78: in function 'init'
	./sile:44: in main chunk
	[C]: in ?
Syntax Error: Document stream is empty
make: *** [Makefile:1381: selfcheck] Error 1
```
cc: @alerque in case he knows the cause of the above.

--- a/pkgs/development/interpreters/lua-5/default.nix
+++ b/pkgs/development/interpreters/lua-5/default.nix
@@ -1,10 +1,11 @@
 # similar to interpreters/python/default.nix
-{ stdenv, lib, callPackage, fetchurl, fetchpatch }:
+{ stdenv, lib, callPackage, fetchurl, fetchpatch, makeBinaryWrapper }:
 
 rec {
   lua5_4 = callPackage ./interpreter.nix {
     sourceVersion = { major = "5"; minor = "4"; patch = "3"; };
     hash = "1yxvjvnbg4nyrdv10bq42gz6dr66pyan28lgzfygqfwy2rv24qgq";
+    makeWrapper = makeBinaryWrapper;
 
     patches = lib.optional stdenv.isDarwin ./5.4.darwin.patch;
   };
@@ -16,6 +17,7 @@ rec {
   lua5_3 = callPackage ./interpreter.nix {
     sourceVersion = { major = "5"; minor = "3"; patch = "6"; };
     hash = "0q3d8qhd7p0b7a4mh9g7fxqksqfs6mr1nav74vq26qvkp2dxcpzw";
+    makeWrapper = makeBinaryWrapper;
 
     patches =
       lib.optionals stdenv.isDarwin [ ./5.2.darwin.patch ];
@@ -29,6 +31,7 @@ rec {
   lua5_2 = callPackage ./interpreter.nix {
     sourceVersion = { major = "5"; minor = "2"; patch = "4"; };
     hash = "0jwznq0l8qg9wh5grwg07b5cy3lzngvl5m2nl1ikp6vqssmf9qmr";
+    makeWrapper = makeBinaryWrapper;
     patches = lib.optional stdenv.isDarwin ./5.2.darwin.patch;
   };
 
@@ -40,6 +43,7 @@ rec {
   lua5_1 = callPackage ./interpreter.nix {
     sourceVersion = { major = "5"; minor = "1"; patch = "5"; };
     hash = "2640fc56a795f29d28ef15e13c34a47e223960b0240e8cb0a82d9b0738695333";
+    makeWrapper = makeBinaryWrapper;
     patches = (lib.optional stdenv.isDarwin ./5.1.darwin.patch)
       ++ [ ./CVE-2014-5461.patch ];
   };

--- a/pkgs/development/interpreters/lua-5/interpreter.nix
+++ b/pkgs/development/interpreters/lua-5/interpreter.nix
@@ -126,6 +126,7 @@ self = stdenv.mkDerivation rec {
   passthru = rec {
     buildEnv = callPackage ./wrapper.nix {
       lua = self;
+      inherit makeWrapper;
       inherit (luaPackages) requiredLuaModules;
     };
     withPackages = import ./with-packages.nix { inherit buildEnv luaPackages;};

--- a/pkgs/tools/typesetting/sile/default.nix
+++ b/pkgs/tools/typesetting/sile/default.nix
@@ -111,7 +111,6 @@ stdenv.mkDerivation rec {
     homepage = "https://sile-typesetter.org";
     changelog = "https://github.com/sile-typesetter/sile/raw/v${version}/CHANGELOG.md";
     platforms = platforms.unix;
-    broken = stdenv.isDarwin;   # https://github.com/NixOS/nixpkgs/issues/23018
     maintainers = with maintainers; [ doronbehar alerque ];
     license = licenses.mit;
   };

